parsl/tests/test_scaling/test_scale_down.py

```diff
@@ -45,51 +48,45 @@ def sleeper(t):
 
 @pytest.mark.local
 def test_scale_out():
+    logger.info("Starting DFK")
     dfk = parsl.dfk()
 
     # Since we have init_blocks = 0, at this point we should have 0 managers
-    print("Executor : ", dfk.executors['htex_local'])
-    print("Before")
-    print("Managers   : ", dfk.executors['htex_local'].connected_managers)
-    print("Outstanding: \n", dfk.executors['htex_local'].outstanding)
+    logger.info(f"Executor: {dfk.executors['htex_local']}")
+    logger.info(f"Managers: {dfk.executors['htex_local'].connected_managers}")
+    logger.info(f"Outstanding: {dfk.executors['htex_local'].outstanding}")
     assert len(dfk.executors['htex_local'].connected_managers) == 0, "Expected 0 managers at start"
 
     fus = [sleeper(i) for i in [3, 3, 25, 25, 50]]
 
+    logger.info("All apps invoked. Waiting for first two to finish.")
+
     for i in range(2):
         fus[i].result()
 
-    # At this point, since we have 1 task already processed we should have atleast 1 manager
-    print("Between")
-    print("Managers   : ", dfk.executors['htex_local'].connected_managers)
-    print("Outstanding: \n", dfk.executors['htex_local'].outstanding)
-    assert len(dfk.executors['htex_local'].connected_managers) == 5, "Expected 5 managers once tasks are running"
-
-    time.sleep(1)
-    print("running")
-    print("Managers   : ", dfk.executors['htex_local'].connected_managers)
-    print("Outstanding: \n", dfk.executors['htex_local'].outstanding)
-    assert len(dfk.executors['htex_local'].connected_managers) == 5, "Expected 5 managers 3 seconds after 2 tasks finished"
-
-    time.sleep(21)
-    print("Middle")
-    print("Managers   : ", dfk.executors['htex_local'].connected_managers)
-    print("Outstanding: \n", dfk.executors['htex_local'].outstanding)
-    assert len(dfk.executors['htex_local'].connected_managers) == 3, "Expected 3 managers before cleaning up"
-
-    for i in range(2, 4):
-        fus[i].result()
-    time.sleep(21)
-    print("Finalizing result")
-    print("Managers   : ", dfk.executors['htex_local'].connected_managers)
-    print("Outstanding: \n", dfk.executors['htex_local'].outstanding)
-    assert len(dfk.executors['htex_local'].connected_managers) == 2, "Expected 2 managers before finishing, lower bound by min_blocks"
+    # At this point, since we have 1 task already processed we should have at least 1 manager
+    logger.info(f"Managers: {dfk.executors['htex_local'].connected_managers}")
+    logger.info(f"Outstanding: {dfk.executors['htex_local'].outstanding}")
+    assert len(dfk.executors['htex_local'].connected_managers) >= 1, "Expected at least one manager once tasks are running"
+
+    # 1s delay here doesn't seem to be long enough for all of the managers to be started and connected in CI.
```

Comment:
I have been very confused about why the assertion would fail on CI. I also tried to tune the time. On my local, I tried >100 times, and none of them failed in its current form. It is not just 1s delay, it already resolves the first two 3-second tasks at this stage, which is already like 4s.

Reply:
Probably CI machine is too slow for that